cmd/newrelic-infra/newrelic-infra.go

```diff
@@ -303,6 +318,34 @@ func initializeAgentAndRun(c *config.Config, logFwCfg config.LogForward) error {
 	}
 
 	aslog.Info("Checking network connectivity...")
+
+	if c.NetProfile {
+		// custom test suit to check endpoint's reachability using different implementation
+		testCases := []func(string, time.Duration, http.RoundTripper, log.Entry) (bool, error){
```

Comment:
Would make sense to move this into the dnstest file?
Also from the naming perspective, I would go with checks instead of tests, wdyt?

Reply:
updated